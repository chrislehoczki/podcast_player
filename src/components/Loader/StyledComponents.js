import styled from "styled-components";

export const Ring = styled.div`
  display: inline-block;
  width: 100%;
  height: 100%;
  opacity: 0.1;
  &:after {
    content: " ";
    display: block;
    width: 55%;
    height: 55%;
    margin: 8px;
    border-radius: 50%;
    border: 6px solid #fff;
    border-color: #fff transparent #fff transparent;
    animation: lds-dual-ring 1.2s linear infinite;
  }
  @keyframes lds-dual-ring {
    0% {
      transform: rotate(0deg);
    }
    100% {
      transform: rotate(360deg);
    }
  }
`;

export const LoaderContainer = styled.div`
  width: ${props => props.size}px;
  height: ${props => props.size}px;
  min-width: ${props => props.size}px;
  min-height: ${props => props.size}px;
  border: ${props => (props.border ? "2px solid black" : 0)};
  outline: 0;
  margin: 20px;
`;
