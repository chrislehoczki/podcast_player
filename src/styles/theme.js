import React from "react";
import styled from "styled-components";

const theme = {
  primary: "#ff7426",
  background: "#3e3e3e"
};

const Loader = styled.div`
  display: inline-block;
  width: 100%;
  height: 100%;
  opacity: 0.1;
  &:after {
    content: " ";
    display: block;
    width: 55%;
    height: 55%;
    margin: 8px;
    border-radius: 50%;
    border: 6px solid #fff;
    border-color: #fff transparent #fff transparent;
    animation: lds-dual-ring 1.2s linear infinite;
  }
  @keyframes lds-dual-ring {
    0% {
      transform: rotate(0deg);
    }
    100% {
      transform: rotate(360deg);
    }
  }
`;

const LoaderContainer = styled.div`
  width: ${props => props.size}px;
  height: ${props => props.size}px;
  min-width: ${props => props.size}px;
  min-height: ${props => props.size}px;
  border: ${props => (props.border ? "2px solid black" : 0)};
  outline: 0;
  margin: 20px;
`;

const PlayButton = (props, onClick) => {
  if (!props.loaded) {
    return (
      <LoaderContainer size={props.size}>
        <Loader />
      </LoaderContainer>
    );
  }
  return <Button {...props} onClick={onClick} />;
};

const Button = styled.button`
  background: url(${props => props.img}) no-repeat center center;
  background-size: 32px;
  width: ${props => props.size}px;
  height: ${props => props.size}px;
  min-width: ${props => props.size}px;
  min-height: ${props => props.size}px;
  filter: invert(1);
  cursor: pointer;
  border-radius: 100%;
  border: ${props => (props.border ? "2px solid black" : 0)};
  outline: 0;
  margin: 20px;
`;

export default theme;

console.log(PlayButton);
export { Button, PlayButton };
