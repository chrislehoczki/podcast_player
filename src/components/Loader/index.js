import React from "react";
import propTypes from "prop-types";

import { Ring, LoaderContainer } from "./StyledComponents";
const Loader = ({ size }) => {
  return (
    <LoaderContainer size={size}>
      <Ring />
    </LoaderContainer>
  );
};

Loader.propTypes = {
  size: propTypes.number
};

export default Loader;
