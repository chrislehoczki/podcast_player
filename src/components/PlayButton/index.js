import React from "react";
import PropTypes from "prop-types";
import { Button } from "../../styles/theme";
import Loader from "../Loader";

const PlayButton = ({
  loaded,
  playing,
  stopPodcast,
  playPodcast,
  podcastId
}) => {
  return !loaded ? (
    <Loader size={60} />
  ) : playing ? (
    <Button
      img={"icons/pause.png"}
      size={60}
      border={true}
      onClick={() => loaded && stopPodcast()}
    />
  ) : (
    <Button
      img={"icons/play.png"}
      size={60}
      border={true}
      onClick={() => playPodcast({ id: podcastId })}
    />
  );
};

PlayButton.propTypes = {
  playPodcast: PropTypes.func,
  stopPodcast: PropTypes.func,
  loaded: PropTypes.bool,
  playing: PropTypes.bool,
  podcastId: PropTypes.string
};

export default PlayButton;
