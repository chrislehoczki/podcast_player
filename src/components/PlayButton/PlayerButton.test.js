import React from "react";
import { shallow, mount } from "enzyme";
import PlayButton from "./index";
import Loader from "../Loader";
import { Button } from "../../styles/theme";

describe("<PlayButton />", () => {
  it("Renders correctly", () => {
    const component = shallow(<PlayButton loaded={false} playing={false} />);
    expect(component).toMatchSnapshot();
  });
  it("Should show a loader and not a button if podcast is not loaded", () => {
    const component = shallow(<PlayButton loaded={false} playing={false} />);
    expect(component.find(Button).exists()).toBeFalsy();
    expect(component.find(Loader)).toExist();
  });
  it("Should show a button and not a loader if podcast is loaded", () => {
    const component = shallow(<PlayButton loaded={true} playing={false} />);
    expect(component.find(Loader).exists()).toBeFalsy();
    expect(component.find(Button)).toExist();
  });
  it("Should show a pause icon if podcast is playing", () => {
    const component = mount(<PlayButton loaded={true} playing={true} />);
    expect(component.find(Button).props().img).toEqual("icons/pause.png");
    expect(component.find(Button)).toHaveStyleRule(
      "background",
      `url(${"icons/pause.png"}) no-repeat center center`
    );
  });
  it("Should show a play icon if podcast is playing", () => {
    const component = mount(<PlayButton loaded={true} playing={false} />);
    expect(component.find(Button).props().img).toEqual("icons/play.png");
    expect(component.find(Button)).toHaveStyleRule(
      "background",
      `url(${"icons/play.png"}) no-repeat center center`
    );
  });
  it("Should call playPodcast when play button is clicked", () => {
    const playPodcast = jest.fn();
    const component = shallow(
      <PlayButton loaded={true} playing={false} playPodcast={playPodcast} />
    );
    component.find(Button).simulate("click");
    expect(playPodcast).toHaveBeenCalledTimes(1);
  });
  it("Should call stopPodcast when play button is clicked", () => {
    const stopPodcast = jest.fn();
    const component = shallow(
      <PlayButton loaded={true} playing={true} stopPodcast={stopPodcast} />
    );
    component.find(Button).simulate("click");
    expect(stopPodcast).toHaveBeenCalledTimes(1);
  });
});
