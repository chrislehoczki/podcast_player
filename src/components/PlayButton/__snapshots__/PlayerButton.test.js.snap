// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PlayButton /> Renders correctly 1`] = `
<Loader
  size={60}
/>
`;
